test/SPIRV/transcoding/RecursiveType.ll

```diff
@@ -0,0 +1,43 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o %t.txt
+; RUN: FileCheck < %t.txt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+%struct.Node = type { %struct.Node addrspace(1)*, i32 }
```

Comment:
Added similar case to the test.
Also I've added support for OpTypeForwardPointer as it's required by SPIR-V specification: https://www.khronos.org/registry/spir-v/specs/1.0/SPIRV.html#_universal_validation_rules (data rules).